rust/src/dataset.rs

```diff
@@ -441,6 +441,85 @@ impl Dataset {
         Self::write_impl(batches, uri, params).await
     }
 
+    pub async fn append_impl(
```

Comment:
Keep the implementation private.
```suggestion
    async fn append_impl(
```